consumer/pdata/common.go

```diff
@@ -244,16 +380,43 @@ func newAttributeMap(orig *[]*otlpcommon.AttributeKeyValue) AttributeMap {
 // assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{...}), actual)
 func (am AttributeMap) InitFromMap(attrMap map[string]AttributeValue) AttributeMap {
 	if len(attrMap) == 0 {
-		*am.orig = []*otlpcommon.AttributeKeyValue(nil)
+		*am.orig = []*otlpcommon.KeyValue(nil)
 		return am
 	}
-	origs := make([]otlpcommon.AttributeKeyValue, len(attrMap))
-	wrappers := make([]*otlpcommon.AttributeKeyValue, len(attrMap))
+	anyVals := make([]otlpcommon.AnyValue, len(attrMap))
+	origs := make([]otlpcommon.KeyValue, len(attrMap))
+	wrappers := make([]*otlpcommon.KeyValue, len(attrMap))
 	ix := 0
 	for k, v := range attrMap {
+		origs[ix].Value = &anyVals[ix]
 		wrappers[ix] = &origs[ix]
 		wrappers[ix].Key = k
-		v.CopyTo(AttributeValue{wrappers[ix]})
+		v.copyTo(&anyVals[ix])
+		ix++
+	}
+	*am.orig = wrappers
+	return am
+}
+
+// InitFromMap overwrites the entire AttributeMap and reconstructs the AttributeMap
+// with values from the given map[string]string.
+//
+// Returns the same instance to allow nicer code like:
+// assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{...}), actual)
+func (am AttributeMap) InitFromAttributeMap(attrMap AttributeMap) AttributeMap {
+	srcLen := attrMap.Len()
+	if srcLen == 0 || attrMap.orig == nil {
+		*am.orig = []*otlpcommon.KeyValue(nil)
+		return am
+	}
+	anyVals := make([]otlpcommon.AnyValue, srcLen)
+	origs := make([]otlpcommon.KeyValue, srcLen)
+	wrappers := make([]*otlpcommon.KeyValue, srcLen)
+	for ix, v := range *attrMap.orig {
```

Comment:
Do we need origs to be a slice? Why not a single orig var whose Value attribute get overwritten?